### ISSUE TITLE
Does not play well with progressive enhancement

### ISSUE THREAD

**Issue by aarongustafson** [gfscott/eleventy-plugin-youtube-embed#66]
How would you feel about me implementing a slightly more progressively enhanced version of the markup? Here are the changes I’ve made to my own copy of Paul’s files:

1. Make the play button a link to play the video.
2. preventDefault for the play button’s onclick event.

This way it will show the poster and play button by default (during loading) and then transparently upgrade to the web component version async.

**Comment by gfscott** [comment]
Yeah, I'm into this! So just to clarify, this would be similar (though not identical) to the "[Pro usage](https://github.com/paulirish/lite-youtube-embed#pro-usage-load-w-js-deferred-aka-progressive-enhancement)" example in Paul's readme, correct?

Are there any changes required to the underlying `lite-youtube-embed` package? Or can it be done purely by updating [the web component markup](https://github.com/gfscott/eleventy-plugin-youtube-embed/blob/main/lib/buildEmbed.js#L34-L36)?

Big fan of your work, by the way! 🤩

**Comment by aarongustafson** [comment]
Awesome, thanks!

To be clear it would replace the PE example. [I put together a PR against Paul’s repo](https://github.com/paulirish/lite-youtube-embed/pull/117) and can copy the code changes to yours in the interim (if you like). It’s basically [one additional line of JavaScript](https://github.com/paulirish/lite-youtube-embed/pull/117/commits/3af89c60b1b9d020904994037d16b360f99e02e7) (to cancel the default event) and [one line of CSS](https://github.com/paulirish/lite-youtube-embed/pull/117/commits/e16db1ae335a9483fb16dcf03a162406721c6b7d) (to make the button display block). Then I’d just tweak your markup template to swap the `button` for `a[href]` like [I did in the PE example in his repo](https://github.com/paulirish/lite-youtube-embed/pull/117/commits/e3c5194273b2e4b60ee1dca12f5c3f13057d4123).

**Comment by gfscott** [comment]
Got it!

One concern: I did make the deliberate choice way back when to simply consume the JS and CSS included with v0.2.0 of the `lite-youtube-embed` package, so as not to maintain a slightly divergent fork of Paul's codebase here.

So my _ideal_ would be to see your changes in JS and CSS merged into the upstream package, but I don't know what our prospects are there. Definitely open to alternatives!

**Comment by aarongustafson** [comment]
Totally with you. Like I said, I have a PR in for these very changes with the package, so maybe we can put a little pressure on Paul for a 👍🏻/👎🏻 on that. I'll ping him.
